### ISSUE TITLE
Tokens not aligned error when spacy < 2.3.0

### ISSUE THREAD

**Issue by HiromuHota** [HazyResearch/fonduer#527]
## Description of the bug

"Tokens are not aligned!" RuntimeError happens at `HocrVisualParser.parse`

## To Reproduce
1. Use the latest commit (a2d097be6f6509214a93246ad8027aeed509ba65)
2. Parse [William_R._King.hocr.txt](https://github.com/HazyResearch/fonduer/files/5427085/William_R._King.hocr.txt) with `HOCRDocPreprocessor` and `HocrVisualParser`

## Expected behavior

Parse without an error

## Error Logs/Screenshots

```
[ERROR] Document William_R._King not added to database, because of parse error: 
('Tokens are not aligned!', 'hocr tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon.\', \'William\', \'R.\', \'King,\', \'of\', \'Alabama,\', \'vice-president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States,\', \'eighth\', \'of\', \'December,\', "1853\'\'"]', 'spacy tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King\', \'?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon\', \'.\', \'William\', \'R.\', \'King\', \',\', \'of\', \'Alabama\', \',\', \'vice\', \'-\', \'president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States\', \',\', \'eighth\', \'of\', \'December\', \',\', \'1853\', "\'", "\'"]')
Traceback (most recent call last):
  File "/Users/hiromu/workspace/fonduer/src/fonduer/parser/parser.py", line 246, in apply
    document.name, document.sentences
  File "/Users/hiromu/workspace/fonduer/src/fonduer/parser/parser.py", line 244, in <listcomp>
    y
  File "/Users/hiromu/workspace/fonduer/src/fonduer/parser/visual_parser/hocr_visual_parser.py", line 134, in parse
    f"spacy tokens: {spacy_tokens}",
RuntimeError: ('Tokens are not aligned!', 'hocr tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon.\', \'William\', \'R.\', \'King,\', \'of\', \'Alabama,\', \'vice-president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States,\', \'eighth\', \'of\', \'December,\', "1853\'\'"]', 'spacy tokens: [\'Who\', \'is\', \'William\', \'Rufus\', \'King\', \'?\', "\'\'", \'Obituary\', \'addresses\', \'on\', \'the\', \'occasion\', \'of\', \'the\', \'death\', \'of\', \'the\', \'Hon\', \'.\', \'William\', \'R.\', \'King\', \',\', \'of\', \'Alabama\', \',\', \'vice\', \'-\', \'president\', \'of\', \'the\', \'United\', \'States\', \':\', \'delivered\', \'in\', \'the\', \'Senate\', \'and\', \'House\', \'of\', \'Representatives\', \'of\', \'the\', \'United\', \'States\', \',\', \'eighth\', \'of\', \'December\', \',\', \'1853\', "\'", "\'"]')
```

## Environment (please complete the following information)
 - OS: macOS 10.15.6
 - PostgreSQL Version: N/A
 - Poppler Utils Version: N/A
 - Fonduer Version: 0.8.4+dev (a2d097be6f6509214a93246ad8027aeed509ba65)
 - spaCy: 2.2.4

## Additional context

This error does not happen when spacy >= 2.3.0

**Comment by HiromuHota** [comment]
The corresponding part in hOCR looks like this:

```
        <span class="ocrx_line" title="bbox 35 349 176 361">
          <span class="ocrx_word" title="bbox 35 349 59 361">Who</span>
          <span class="ocrx_word" title="bbox 62 349 70 361">is</span>
          <span class="ocrx_word" title="bbox 73 349 112 361">William</span>
          <span class="ocrx_word" title="bbox 115 349 143 361">Rufus</span>
          <span class="ocrx_word" title="bbox 146 349 176 361">King?</span>
        </span>
        <span class="ocrx_line" title="bbox 35 362 579 374">
          <span class="ocrx_word" title="bbox 35 362 40 374">''</span>
          <span class="ocrx_word" title="bbox 43 362 85 374">Obituary</span>
          <span class="ocrx_word" title="bbox 88 362 134 374">addresses</span>
          <span class="ocrx_word" title="bbox 137 362 149 374">on</span>
          <span class="ocrx_word" title="bbox 152 362 167 374">the</span>
```

When I erased one of the two single-quotes (`''`), the error disappeared.

**Comment by HiromuHota** [comment]
spaCy v2.2.4 on Mar 12 released and v2.3.0 released Jun 16.
I suspect one of the changes between these dates changed how `gold.align` works. https://github.com/explosion/spaCy/commits/master/spacy/gold.pyx

**Comment by HiromuHota** [comment]
I think https://github.com/explosion/spaCy/pull/5334 is the one, which "Remove alignment normalizations beyond whitespace and lowercasing".

```
_ALIGNMENT_NORM_MAP = [("``", "'"), ("''", "'"), ('"', "'"), ("`", "'")]


def _normalize_for_alignment(tokens):
    tokens = [w.replace(" ", "").lower() for w in tokens]
    output = []
    for token in tokens:
        token = token.replace(" ", "").lower()
        for before, after in _ALIGNMENT_NORM_MAP:
            token = token.replace(before, after)
        output.append(token)
    return output
```

As of v2.2.4, two single quotes `''` were normalized to a single quote `'`.
